src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -158,15 +178,9 @@ public static Set<Specialisation> parseSpecialisations(String specialisations)
         if (trimmedSpecialisations.isEmpty()) {
             return new HashSet<>();
         }
-        String[] specialisationNames = trimmedSpecialisations.split(",");
+
         Set<Specialisation> specialisationSet = new HashSet<>();
-        for (String specialisationName : specialisationNames) {
-            String specialisationNameTrimmed = specialisationName.trim();
-            if (!Specialisation.isValidSpecialisationName(specialisationNameTrimmed)) {
-                throw new ParseException(Specialisation.MESSAGE_CONSTRAINTS);
-            }
-            specialisationSet.add(new Specialisation(specialisationNameTrimmed));
-        }
+        addSpecialisationsFromString(trimmedSpecialisations, specialisationSet);
```

Comment:
Applying SLAP principle, good work, makes the main parseSpecialisations function more readable.